src/Forms/GridField/GridFieldDetailForm.php

```diff
@@ -129,7 +129,8 @@ public function handleItem($gridField, $request)
     protected function getItemRequestHandler($gridField, $record, $requestHandler)
     {
         $class = $this->getItemRequestClass();
-        $this->extend('updateItemRequestClass', $class, $gridField, $record, $requestHandler);
+        $assignedClass = $this->itemRequestClass;
+        $this->extend('updateItemRequestClass', $class, $gridField, $record, $requestHandler, $assignedClass);
```

Comment:
I'm not super familiar with this class, but to me it feels a bit weird that we have a `updateItemRequestClass` hook in a method call `getItemRequestHandler()`. 

If we need to allow extension to alter the ItemRequestClass wouldn't that be better handled in `getItemRequestClass()`.

On a different topic, it's not clear at all here that `$class` is being passed by reference, which makes the code a bit confusing.